plugins/launcher_service_plugin/include/eosio/launcher_service_plugin/launcher_service_plugin.hpp

```diff
@@ -0,0 +1,357 @@
+      bool             reverse = false;
+      bool             show_payer = false; // show RAM pyer
+    };
+
+   struct cluster_id_param {
+      int              cluster_id = 0;
+   };
+   struct node_id_param {
+      int              cluster_id = 0;
+      int              node_id = 0;
+   };
+
+   struct start_node_param {
+      int              cluster_id = 0;
+      int              node_id = 0;
+      std::string      extra_args;
+   };
+   struct stop_node_param {
+      int              cluster_id = 0;
+      int              node_id = 0;
+      int              kill_sig = 15;
+   };
+
+   struct send_raw_param { // send abitrary string or json data to one node
+      int              cluster_id = 0;
+      int              node_id = 0;
+      std::string      url; // suburl of endpoint
+      std::string      string_data; // either string or json
+      fc::variant      json_data;
+   };
+}
+
+/**
+ *  This is a template plugin, intended to serve as a starting point for making new plugins
+ */
+class launcher_service_plugin : public appbase::plugin<launcher_service_plugin> {
+public:
+   launcher_service_plugin();
+   virtual ~launcher_service_plugin();
+
+   APPBASE_PLUGIN_REQUIRES()
+   virtual void set_program_options(options_description&, options_description& cfg) override;
+
+   void plugin_initialize(const variables_map& options);
+   void plugin_startup();
+   void plugin_shutdown();
+
+   // cluster related calls
+   fc::variant launch_cluster(launcher_service::cluster_def cluster_def);
+   fc::variant stop_all_clusters();
+   fc::variant stop_cluster(launcher_service::cluster_id_param);
+   fc::variant clean_cluster(launcher_service::cluster_id_param);
+   fc::variant start_node(launcher_service::start_node_param);
+   fc::variant stop_node(launcher_service::stop_node_param);
+
+   // wallet related calls
+   fc::variant generate_key(launcher_service::generate_key_param);
+   fc::variant import_keys(launcher_service::import_keys_param);
+
+   // queries
+   fc::variant get_info(launcher_service::node_id_param);
+   fc::variant get_block(launcher_service::get_block_param);
+   fc::variant get_block_header_state(launcher_service::get_block_param);
+   fc::variant get_account(launcher_service::get_account_param);
+   fc::variant get_code_hash(launcher_service::get_account_param);
+   fc::variant get_cluster_info(launcher_service::cluster_id_param);
+   fc::variant get_cluster_running_state(launcher_service::cluster_id_param);
+   fc::variant get_protocol_features(launcher_service::node_id_param);
+   fc::variant verify_transaction(launcher_service::verify_transaction_param);
+   fc::variant get_table_rows(launcher_service::get_table_rows_param);
+   fc::variant get_log_data(launcher_service::get_log_data_param);
+
+   // transactions
+   fc::variant create_bios_accounts(launcher_service::create_bios_accounts_param);
+   fc::variant create_account(launcher_service::new_account_param_ex);
```

Comment:
Our tests should not be using the `eosio.system` contract. This means there should not any reason to buy RAM or stake resources for CPU/NET. If a test needs to limit resources of an account it should set the RAM/CPU/NET limits of the account explicitly using the appropriate `eosio.bios` action. However, normally a test would not need to change the limits of an account from its default of unlimited.

Reply:
removed create account related PRC